### ISSUE TITLE
Debugger cannot start:  ERR timeout after 1000 ms: Error: timeout after 1000 ms

### ISSUE THREAD

**Issue by ctataryn** [elixir-lsp/vscode-elixir-ls#114]
### Environment

Most of this can be filled out by running the VSCode command (by default bound to Ctrl+Shift+P) "ElixirLS: Copy Debug Info"

  * Elixir & Erlang versions (elixir --version): Erlang/OTP 22 [erts-10.7] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1]
Elixir 1.9.4 (compiled with Erlang/OTP 20)

  * VSCode ElixirLS version: 0.4.0
  * Operating System Version: win32 10.0.18363
  

### Troubleshooting

- [ x] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [ x] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor

### Crash report template

After upgrading from v0.4.0 -> v0.5.0 debugging has stopped working. The Developer's Console states:

```
  ERR timeout after 1000 ms: Error: timeout after 1000 ms
    at t.RawDebugSession.handleErrorResponse (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:3160:190)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:3159:451
    at async t.RawDebugSession.shutdown (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:3157:536)
```

Using `git bisect` I was able to track it down to this specific commit:

https://github.com/elixir-lsp/vscode-elixir-ls/commit/97066cefed8cf4111b63241ff5d931b2f3e39614

My guess is, perhaps on Windows machines, the workspace scanner times things out? The strange thing is, even on a relatively empty project (i.e. generated from `mix phx.new myproj`) it still does this.

**Comment by axelson** [comment]
@ctataryn thanks for this! So with the change in #115 you don't get the `ERR timeout`?

@jayjun do you know of any other workarounds we could use to avoid `**/*.ex`? Although perhaps the better fix would be to change the command to return a simple error if the extension isn't started.

For reference #107 was originally created to improve the Copy Debug Info command: https://github.com/elixir-lsp/vscode-elixir-ls/pull/70#issuecomment-640395218

**Comment by jayjun** [comment]
The problem is there is no opportunity to return an error. _Copy Debug Info_ is registered as globally available by [`contributes.commands`](https://code.visualstudio.com/api/references/contribution-points#contributes.commands)  at launch but the extension isn’t started (thus registering any handlers) until an Elixir file is opened.

However, you can conditionally enable it with a [`when` clause](https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts). This should do it.

```diff
"commands": [
  {
    "command": "extension.copyDebugInfo",
    "title": "ElixirLS: Copy Debug Info",
+   "when": "editorLangId == elixir"
  }
]
```
The drawback is users must be instructed to have an Elixir file opened to access that command, say before filing a bug report.

**Comment by ctataryn** [comment]
> thanks for this! So with the change in #115 you don't get the ERR timeout?

That's correct.  The commit before this one works, and everyone afterwards fails with the timeout.

**Comment by axelson** [comment]
> However, you can conditionally enable it with a [`when` clause](https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts). This should do it.

Okay, that seems like a good approach :+1:

**Comment by axelson** [comment]
Fixed in #115
